sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2446,10 +2446,11 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
 
   /**
    * Type to keep track of table clauses:
-   * (partitioning, bucketSpec, properties, options, location, comment).
+   * (partTransforms, partCols, bucketSpec, properties, options, location, comment, serde).
```

Comment:
nit: let's add more comments:
```
   * Note: partition transforms are based on existing table schema definition. It can be simple
   * column names, or functions like `year(date_col)`. partition columns are column names with data
   * types like `i INT`, which should be appended to the existing table schema.
```